src/plugins/intel_cpu/src/emitters/snippets/x64/jit_parallel_loop_emitters.cpp

```diff
@@ -0,0 +1,417 @@
+    push_reg_on_stack(aux_reg, GET_OFF_PARALLEL_LOOP_ARGS(mem_ptrs));
+
+    h->mov(aux_reg, reinterpret_cast<uintptr_t>(ParallelLoopExecutor::execute));
+    h->mov(abi_param1, reinterpret_cast<uintptr_t>(m_executor.get()));
+    h->mov(abi_param2, h->rsp);
+
+    binary_call_reg_spiller.rsp_align(get_callee_saved_reg().getIdx());
+    // Note: we will return from this call only when the parallel region is finished (return from
+    // jit_parallel_loop_end_emitter)
+    h->call(aux_reg);
+    binary_call_reg_spiller.rsp_restore();
+
+    // Restore data ptrs with applied finalization offsets
+    for (size_t i = 0; i < mem_ptr_regs_idxs.size(); ++i) {
+        h->mov(Reg64(mem_ptr_regs_idxs[i]), h->qword[h->rsp + call_args_size + i * sizeof(uintptr_t*)]);
+    }
+    h->add(h->rsp, reserved_stack_size);
+    binary_call_reg_spiller.postamble();
+
+    h->jmp(*loop_end_label, CodeGenerator::T_NEAR);
+}
+
+void jit_parallel_loop_begin_emitter::emit_parallel_region_initialization(
+    const std::vector<Xbyak::Reg>& regs_to_restore) const {
+    h->L(*loop_preamble_label);
+
+    std::set<snippets::Reg> loop_premble_spill;
+    // todo: we don't have to spill all calle+saved_regs, only the ones that will be used in the loop's body
+    for (auto i : get_callee_saved_reg_idxs()) {
+        loop_premble_spill.insert({snippets::RegType::gpr, i});
+    }
+    m_parallel_section_reg_spiller->preamble(loop_premble_spill);
+
+    // Note: some of mem_ptr_regs_idxs might coincide with abi_param_2.
+    // abi_param_1 is always reserved for runtime parameters storage,
+    // so it can't coincide with any of mem_ptr_regs_idxs.
+    std::optional<size_t> abi_param2_collision_index;
+    for (size_t i = 0; i < mem_ptr_regs_idxs.size(); ++i) {
+        auto reg_to_restore = Reg64(mem_ptr_regs_idxs[i]);
+        OPENVINO_ASSERT(
+            std::find(regs_to_restore.begin(), regs_to_restore.end(), reg_to_restore) == regs_to_restore.end(),
+            "Expected to restore all registers except for mem_ptr_regs_idxs");
+        if (reg_to_restore == abi_param2) {
+            abi_param2_collision_index = i;
+        } else {
+            h->mov(reg_to_restore, h->ptr[abi_param2 + i * sizeof(uintptr_t*)]);
+        }
+    }
+    if (abi_param2_collision_index.has_value()) {
+        const auto collision_idx = abi_param2_collision_index.value();
+        h->mov(abi_param2, h->ptr[abi_param2 + collision_idx * sizeof(uintptr_t*)]);
+        OPENVINO_ASSERT(work_amount_reg_idx != static_cast<size_t>(abi_param2.getIdx()),
+                        "Unexpected collision: the same reg is allocated for work_amount and memory pointer");
+    }
+    h->mov(Reg64(work_amount_reg_idx), abi_param1);
+
+    const auto& aux_reg = get_call_address_reg();
+    h->mov(aux_reg, reinterpret_cast<uintptr_t>(m_common_registers_buffer.data()));
+    EmitABIRegSpills::load_regs_from_memory(h, regs_to_restore, aux_reg, m_common_registers_buffer.size());
```

Comment:
Please double check: can `get_call_address_reg()` return register which is equal to some reg from `regs_to_restore` or to `abi_param2`? Or is calling `get_aux_gpr` safer here?

Reply:
`get_call_address_reg` is taken from `aux_gpr_idxs`, and it never coincides with `regs_to_restore`. I added an assertion which checks that just in case